frappe/model/document.py

```diff
@@ -403,7 +403,9 @@ def get_doc_before_save(self):
 	def has_value_changed(self, fieldname):
 		'''Returns true if value is changed before and after saving'''
 		previous = self.get_doc_before_save()
-		return previous.get(fieldname)!=self.get(fieldname) if previous else True
+		if previous is not None and previous.get(fieldname)!=self.get(fieldname):
```

Comment:
Your fix is incorrect. It would return `False` for any field in a new doc (set or unset). 
Current design is simple: if it is a new doc, `has_value_changed` always returns `True`.

Perhaps `has_value_changed` needn't be used when doc is new.

Reply:
My scenario of using the function is to check whether the docfield was changed and trigger the relevant actions. If the field in new doc is not set, the actions need not be triggered.

Reply:
Maybe you should use `doc.is_new()` first?